wrappers/winuser.go

```diff
@@ -240,17 +240,19 @@ func GetForegroundWindow() syscall.Handle {
 	return syscall.Handle(r1)
 }
 
-func GetWindowText(hwnd syscall.Handle, buffer *uint16, maxCount int32) (int, error) {
+func GetWindowText(hwnd syscall.Handle, buffer *uint16, maxCount int32) (int32, error) {
 	r1, _, e1 := syscall.Syscall(
 		procGetWindowTextW.Addr(),
 		3,
 		uintptr(hwnd),
 		uintptr(unsafe.Pointer(buffer)),
 		uintptr(maxCount))
-	if e1 != ERROR_SUCCESS {
-		return int(r1), e1
+	if r1 == 0 {
+		if e1 != ERROR_SUCCESS {
+			return 0, e1
+		}
```

Comment:
This one should still return `syscall.EINVAL` for `ERROR_SUCCESS`.  It's only `GetWindowTextLength` that needs the special handling.

Reply:
The window can have empty text. And in this case GetWindowText and GetWindowTextLength returns 0. But this is not error. 

I checked this and generally:
- in case of success GetWindowText, GetWindowTextLength do NOT set LastError value. 
LastError is set only in the case of error (for example when handle is invalid)

Can you have any idea how to handle errors in this situation ?

Reply:
Good catch, I was mistaken on that one.  In terms of how to fix it, just call `SetLastError(ERROR_SUCCESS)` before calling `GetWindowText`.  You'll need to add `SetLastError` to winbase.go, since we don't have it yet.

Reply:
However, it looks like that SetLastError(ERROR_SUCCESS) is not needed. I checked behavior of GetWindowText, GetWindowTextLength in C++ and LastError was not set. But in go it is. I think that syscall.Syscall always sets SetLastError(ERROR_SUCCESS) before system call (see runtime\sys_windows_386.s)